picasso/src/main/java/com/squareup/picasso/BitmapHunter.java

```diff
@@ -570,4 +582,33 @@ private static boolean shouldResize(boolean onlyScaleDown, int inWidth, int inHe
       int targetWidth, int targetHeight) {
     return !onlyScaleDown || inWidth > targetWidth || inHeight > targetHeight;
   }
+
+ static int getExifRotation(int orientation) {
+    switch (orientation) {
+      case ORIENTATION_ROTATE_90:
+      case ORIENTATION_TRANSPOSE:
+        return 90;
+      case ORIENTATION_ROTATE_180:
+      case ORIENTATION_FLIP_VERTICAL:
```

Comment:
Double checked it and it works. There's perhaps a single matrix operation that could achieve this. I'll look into it.